src/main/java/com/refinedmods/refinedstorage/integration/jei/GridRecipeTransferHandler.java

```diff
@@ -33,55 +38,107 @@
 
     @Override
     public IRecipeTransferError transferRecipe(@Nonnull GridContainer container, Object recipe, @Nonnull IRecipeLayout recipeLayout, @Nonnull PlayerEntity player, boolean maxTransfer, boolean doTransfer) {
-        IGrid grid = container.getGrid();
+        JEIIngredientTracker tracker = trackItems(container, recipeLayout, player);
 
-        if (doTransfer) {
-            LAST_TRANSFER_TIME = System.currentTimeMillis();
+        if (tracker.hasMissing()) {
+            if (doTransfer && Screen.hasControlDown()) {
+                tracker.getCraftingRequests().forEach((id, count) -> {
+                    RS.NETWORK_HANDLER.sendToServer(new GridCraftingPreviewRequestMessage(id, count, Screen.hasShiftDown(), false));
```

Comment:
How does this work, what happens if multiple responses are retrieved? I don't see any logic handling that.

Reply:
I was fully prepared to rewrite the preview screen for this or introduce some for of combined preview.
But this just works. The preview for the first craft pops up and then a preview for the next one.

Reply:
Why does that work?